src/main/java/org/example/game2048/movements/Movement.java

```diff
@@ -1,172 +1,109 @@
 package org.example.game2048.movements;
-import org.example.game2048.point.Point;
 
+import org.example.game2048.point.Point;
+import java.util.ArrayList;
 import java.util.Comparator;
 import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.stream.Collectors;
-import java.util.stream.IntStream;
 
 public class Movement {
 
-    private static List<Point> TEMPORARY_LIST;
-    private static final ComparatorX COMPARATOR_X = new ComparatorX();
-    private static final ComparatorY COMPARATOR_Y = new ComparatorY();
-    private static int firstPlace, secondPlace, thirdPlace, fourthPlace;    // indexes of main places for numbers on board
-    private static int threeElementsFirst, threeElementsSecond, threeElementsThird, auxiliaryCoordinateIndex;   // auxiliary variables
-    private static int forTwoElementsRowWithoutChangesFirstPlace, forTwoElementsRowWithoutChangesSecondPlace;   // auxiliary variables
-
-    //  x0  x1  x2   x3
-    //  [2]  [2]  [ ]  [2] y0
-    //  [ ]  [4]  [4]  [ ] y1
-    //  [ ]  [2]  [ ]  [ ] y2
-    //  [2]  [ ]  [3]  [ ] y3
-
-    public static void eachRowProcessing (CoordinateGetter rowGetter,
-                                          CoordinateSetter rowSetter,
-                                          Comparator<Point> comparator,
-                                          CoordinateGetter coordinateGetter,
-                                          CoordinateSetter coordinateSetter) {
-        IntStream.range(0, 4).forEach(rowIndex -> {
-
-            List<Point> certainLineOfGameField = TEMPORARY_LIST.stream()
-                    .filter(point -> rowGetter.getCoordinate(point) == rowIndex)
+    public static List<Point> eachRowAndColumnProcessing(List<Point> fullGrid,
```

Comment:
nie jestem fanem tych wszystkich static i synchronized ale to juz wiesz :)

Reply:
gdzies mi umknelo kiedy wywaliles pola z klasy (albo ich nigdy nie bylo i cos mi sie wydawalo ?) 
w kazdym badz razie jezeli ta klasa nie przechowuje zadnego stanu to static bedzie ok :)